### DISCUSSION TITLE
Windows devs required to help pointpoint vsgXchange regresion

### DISCUSSION THREAD

**Discussion by robertosfield** [vsg-dev/VulkanSceneGraph#716] [General]
Hi All,

@rainergericke has reported an issue he's having with latest VSG/vsgXchange master - it crashes when trying to load models via assimp, the isuee is on the vsgXchange Issue tracker:

https://github.com/vsg-dev/vsgXchange/issues/147

I have tried reproduce the issue under Linux, and reviewed the code but neither Rainer or myself has come up with a possible cause of regression.   So I'm looking for Windows users to test out the latest VSG/vsgXciange and see if you can load .obj, .gltg etc. models.

My hope is with more data points of success/failure with different Visual Studio/MinGW/Cygwin build combinations we might be a hint to root cause of the problem.

Loading a model that relies upon vsgXchange::assimp to load it should be sufficient to trigger the crash if it's going to happen, i.e.:

```
vsgviewer cow.obj
vsgconv DamagedHelmet.gltf helmet.vsgb
```

Whether it works or not could you post your findings here, along with the build tools you used.

Thanks.
Robert.

**Comment by Mikalai** [comment]
Hello Robert,

I have tried the latest source code and I also have an issue now. And there is a stack overflow.
```
 	vsgconvd.exe!__chkstk() Line 109	Unknown
>	vsgconvd.exe!<lambda>() Line 32134	C++
 	vsgconvd.exe!vsg::createPhysicsBasedRenderingShaderSet(vsg::ref_ptr<vsg::Options const> options) Line 373	C++
 	vsgconvd.exe!SceneConverter::getOrCreatePbrShaderSet() Line 216	C++
 	vsgconvd.exe!SceneConverter::convert(const aiMaterial * material, vsg::DescriptorConfigurator & convertedMaterial) Line 321	C++
 	vsgconvd.exe!SceneConverter::visit(const aiScene * in_scene, vsg::ref_ptr<vsg::Options const> in_options, const vsg::Path & ext) Line 746	C++
 	vsgconvd.exe!vsgXchange::assimp::Implementation::read(const vsg::Path & filename, vsg::ref_ptr<vsg::Options const> options) Line 1021	C++
 	vsgconvd.exe!vsgXchange::assimp::read(const vsg::Path & filename, vsg::ref_ptr<vsg::Options const> options) Line 83	C++
 	vsgconvd.exe!vsg::CompositeReaderWriter::read(const vsg::Path & filename, vsg::ref_ptr<vsg::Options const> options) Line 47	C++
 	vsgconvd.exe!vsg::read::__l2::<lambda>() Line 32	C++
 	vsgconvd.exe!<lambda_6459c8d8a3e1099ca5069970398c73b5>::operator()<vsg::ref_ptr<vsg::LoadedObject>>(vsg::ref_ptr<vsg::LoadedObject> load) Line 71	C++
 	vsgconvd.exe!vsg::SharedObjects::share<vsg::LoadedObject,<lambda_6459c8d8a3e1099ca5069970398c73b5>>(vsg::ref_ptr<vsg::LoadedObject> & object, vsg::read::__l5::<lambda_6459c8d8a3e1099ca5069970398c73b5> init) Line 91	C++
 	vsgconvd.exe!vsg::read(const vsg::Path & filename, vsg::ref_ptr<vsg::Options const> options) Line 74	C++
 	vsgconvd.exe!main(int argc, char * * argv) Line 341	C++
 	[External Code]	

```

Probably this array which is allocated on stack is too big for VC++ compiler without extra compile flags
```
static auto pbr_ShaderSet = []() {
uint8_t data[] = {
35, 118, 115, 103, 97, 32, 49, 46, 48, 46, 51, 10, 82, 111, 111, 116, 32, 105, 100, 61, 49, 32, 118, 115, 103, 58, 58, 83, 104, 97, 100, 101,

```
When I've made data static error was fixed and vsgview and vsgconv started to work fine.
```
static auto pbr_ShaderSet = []() {
static uint8_t data[] = {
```

**Reply by robertosfield** [reply]
ooo there's an obscure error, thanks for spotting this and provide a possible fix.  I'll apply this to vsgXchange and check it in.

**Comment by robertosfield** [comment]
@Mikalai I have applied your suggestion to vsgXchange's cpp writer:

https://github.com/vsg-dev/vsgXchange/commit/b37861ba6db205fb05bc92ac679aa84b05d17ad7

And have updated the VulkanSceneGraph's built in shaders wit commits : 7924ac96718b90398131ae764bec828abbe2fdba & 2e1ddb9e4d86f61072ae512665fc640c7a8dc01b.

I'd appreciate testing of the latest VulkanSceneGraph and vsgXchange master, so we can close this Issue. Thanks.

**Reply by Mikalai** [reply]
Latest sources looks good. vsgconv and vsgviewer works fine.

**Reply by robertosfield** [reply]
Excellent thanks for the testing and the very helpful suggestion.

**Reply by rainergericke** [reply]
Thanks also from me, it works again perfectly.!
